lib/common/graph.js

```diff
@@ -178,20 +178,12 @@ class ConcertoGraphVisitor extends DiagramVisitor {
      * @protected
      */
     visitClassDeclaration(classDeclaration, parameters) {
-        parameters.stack ??= [];
-        parameters.stack.push(classDeclaration.getFullyQualifiedName());
         parameters.graph.addVertex(classDeclaration.getFullyQualifiedName());
 
         if (classDeclaration.getSuperType()){
             parameters.graph.addEdge(classDeclaration.getFullyQualifiedName(), classDeclaration.getSuperType());
-            // this "if" block adds the types that extend the Super type
-            if(!parameters.createDependencyGraph && classDeclaration.getSuperType() !== 'concerto@1.0.0.Concept') {
-                parameters.graph.addVertex(classDeclaration.getSuperType());
-                parameters.graph.addEdge(classDeclaration.getSuperType(), classDeclaration.getFullyQualifiedName());
-            }
```

Comment:
This helps resolve the issue when a dependency points to its super type
e.g
`pii extends Decorator` should resolve to
`pii --> Decorator`

Reply:
We can't remove this code as it will be a breaking change.

https://github.com/accordproject/concerto-codegen/pull/119/files#diff-0dd19916e3ac1be876b815280030848de979150b7385190b1219996f4f38f735L188
The parameter was designed to allow clients to opt-in to the correct behaviour.

Reply:
Can you add a comment to the code that better explains this, please?